File: BOL/Customer.cs
namespace BOL;
public class Customer
{
    public int Id{get;set;}
    public string FirstName{get;set;}
    public string LastName{get;set;}
    public string Email{get;set;}
    public string Password{get;set;}
    public string Mobile{get;set;}
    public DateTime RegisterDate{get;set;}

    public Customer(int id,string firstName,string lastName,string email,string password,string mobile,string registerDate){
        this.Id=id;
        this.FirstName=firstName;
        this.LastName=lastName;
        this.Email=email;
        this.Password=password;
        this.Mobile=mobile;
        this.RegisterDate=DateTime.Parse(registerDate);
    }

}

File: Backend/gharKaDabba/src/main/java/com/app/entities/Customer.java
package com.app.entities;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.CollectionTable;
import javax.persistence.Column;
import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonManagedReference;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "customers")
@Getter
@Setter
@NoArgsConstructor
public class Customer extends UserEntity {

	public Customer(String firstName, String lastName, String email, String mobile) {
		super(firstName, lastName, email, mobile);

	}

	@ElementCollection(fetch = FetchType.EAGER)
	@CollectionTable(name = "delivery_addresses")
	@Column(name = "cust_id")
	private Set<Address> deliveryAddress = new HashSet<Address>();

	@JsonManagedReference
	@OneToMany(mappedBy = "customer", cascade = CascadeType.ALL, orphanRemoval = true)
	private Set<Order> orders;

	@JsonManagedReference
	@OneToMany(mappedBy = "customer", cascade = CascadeType.ALL, orphanRemoval = true)
	private Set<CustomerPlanSubscription> plans = new HashSet<>();

}
